src/otaclient/grpc/api_v2/main.py

```diff
@@ -79,9 +92,18 @@ async def _grpc_server_launcher():
         logger.info(f"launch grpc API server at {_address_info}")
         await server.start()
         try:
-            await server.wait_for_termination()
+            # Wait for the stop event without busy polling
+            await monitor_stop_event()
         finally:
+            # Ensure server gets stopped properly on exceptions
+            logger.info("stopping grpc API server...")
             await server.stop(1)
             thread_pool.shutdown(wait=True)
+            logger.info("grpc API server stopped")
 
     asyncio.run(_grpc_server_launcher())
+
+    # Keep the process alive even after the gRPC server stops by stop_server_event
+    logger.info("gRPC server has stopped, but keeping the process alive")
+    while client_update_control_flags.stop_server_event.is_set():
+        time.sleep(1)
```

Comment:
As described in the comments, keep sleeping the gRPC process without killing once the server is stopped. Because main process will do shutdown when the process is terminated.